pkg/store/bucket.go

```diff
@@ -62,6 +62,8 @@ import (
 	"github.com/thanos-io/thanos/pkg/tracing"
 )
 
+type StoreDataType string
```

Comment:
Nit. Is it necessary to use string type here? int is probably fine and we can just do iota.

Reply:
true, I originally thought of printing the StoreDataType but not anymore. I'll change it to iota